File: admin/src/dtos/entry.ts
import moment from 'moment'
import { Field, StationInfo } from './station'

export interface BaseEntry {
  id: string
  state: boolean
  stateKind?: StateKind
  amount?: number
  note?: string
  crew?: string
  date: moment.Moment
}

export interface NetworkEntry extends BaseEntry {
  field: string
  station: string
}

export enum StateKind {
  broken = 'broken',
  tooLittle = 'tooLittle',
  other = 'other',
}

export interface EventEntry {
  date: moment.Moment
  firstAid: number
  search: number
}

export interface Entry extends BaseEntry {
  station: StationInfo
  field: Field
}

File: android/app/src/main/java/de/tjarksaul/wachmanager/dtos/Entry.kt
package de.tjarksaul.wachmanager.dtos

import android.os.Parcelable
import kotlinx.android.parcel.Parcelize

@Parcelize
data class Entry(
    val id: String?,
    val field: String,
    val station: String,
    val state: Boolean?,
    val stateKind: StateKind?,
    val amount: Int?,
    val note: String?,
    val date: String // todo: date
) : Parcelable

